src/tests/ftest/server/daos_server_restart.py

```diff
@@ -0,0 +1,218 @@
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+from __future__ import print_function
+
+from apricot import TestWithServers
+from server_utils import ServerFailed
+from daos_utils import DaosCommand
+from command_utils import CommandFailure
+
+
+class DaosServerTest(TestWithServers):
+    """Daos server tests.
+
+    Test Class Description:
+        Test to verify that the daos_io_server, daos_server starts/stops
+        and reformat.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a DaosServerTest object."""
+        super(DaosServerTest, self).__init__(*args, **kwargs)
+
+    def restart_daos_server(self, reformat=True):
+        """method to perform server stop and start.
+        Args:
+            reformat (bool): always reformat storage, could be destructive.
+
+        """
+        self.log.info("=Restart daos_server, server stop().")
+        try:
+            self.server_managers[0].stop()
+        except ServerFailed as error:
+            self.fail(
+                "##Failed on restart daos_server, stopping server, %s", error)
+        self.log.info("=Restart daos_server, prepare().")
+        try:
+            self.server_managers[0].prepare()
+        except ServerFailed as error:
+            self.fail(
+                "##Failed on restart daos_server, prepare(), %s", error)
+        self.log.info("=Restart daos_server, detect_format_ready().")
+        try:
+            self.server_managers[0].detect_format_ready()
+        except ServerFailed as error:
+            self.fail(
+                "##Failed on restart daos_server, detect_format_ready(), %s",
+                error)
+        self.log.info("=Restart daos_server, dmg storage_format.")
+        self.server_managers[0].dmg.exit_status_exception = False
+        try:
+            self.server_managers[0].dmg.storage_format(reformat)
+        except ServerFailed as error:
+            self.fail(
+                "##Failed on restart daos_server, dmg.storage_format(), %s",
+                error)
+        self.log.info("=Restart daos_server, detect_io_server_start().")
+        try:
+            self.server_managers[0].detect_io_server_start()
+        except ServerFailed as error:
+            self.fail(
+                "##Failed on restart daos_server, .detect_io_server_start(),"
+                " %s", error)
+
+    def restart_daos_io_server(self, force=True):
+        """method to perform io_server stop and start by dmg.
+        Args:
+            force (bool): Force to stop the daos io_server.
+            Defaults to True.
+
+        """
+        try:
+            self.server_managers[0].dmg.system_stop(force)
+        except CommandFailure as error:
+            self.fail(
+                "##Failed stopping daos io-server by dmg, %s", error)
+        try:
+            self.server_managers[0].dmg.system_start()
```

Comment:
Again, here you are only staring/stopping 1 server.

Reply:
same as above.